sycl/plugins/unified_runtime/ur/adapters/hip/device.cpp

```diff
@@ -0,0 +1,999 @@
+        // respect to other CPUs and GPUs in the system
+        Value |= UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_ATOMIC_CONCURRENT_ACCESS;
+      }
+    }
+    return ReturnValue(Value);
+  }
+  case UR_DEVICE_INFO_USM_CROSS_SHARED_SUPPORT: {
+    // from cl_intel_unified_shared_memory:
+    // "The cross-device shared memory access capabilities apply to any shared
+    // allocation associated with this device, or to any shared memory
+    // allocation on another device that also supports the same cross-device
+    // shared memory access capability."
+    //
+    // query if/how the device can access managed memory associated to other
+    // devices
+    ur_device_usm_access_capability_flags_t Value = {};
+    if (getAttribute(hDevice, hipDeviceAttributeManagedMemory)) {
+      // the device can allocate managed memory on this system
+      Value |= UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_ACCESS;
+    }
+    if (getAttribute(hDevice, hipDeviceAttributeConcurrentManagedAccess)) {
+      // all devices with the CU_DEVICE_ATTRIBUTE_CONCURRENT_MANAGED_ACCESS
+      // attribute can coherently access managed memory concurrently with the
+      // CPU
+      Value |= UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_CONCURRENT_ACCESS;
+    }
+    if (getAttribute(hDevice, hipDeviceAttributeComputeCapabilityMajor) >= 6) {
+      // compute capability 6.x introduces operations that are atomic with
+      // respect to other CPUs and GPUs in the system
+      if (Value & UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_ACCESS)
+        Value |= UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_ATOMIC_ACCESS;
+      if (Value & UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_CONCURRENT_ACCESS)
+        Value |= UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_ATOMIC_CONCURRENT_ACCESS;
+    }
+    return ReturnValue(Value);
+  }
+  case UR_DEVICE_INFO_USM_SYSTEM_SHARED_SUPPORT: {
+    // from cl_intel_unified_shared_memory:
+    // "The shared system memory access capabilities apply to any allocations
+    // made by a system allocator, such as malloc or new."
+    //
+    // query if/how the device can access pageable host memory allocated by the
+    // system allocator
+    ur_device_usm_access_capability_flags_t Value = {};
+    if (getAttribute(hDevice, hipDeviceAttributePageableMemoryAccess)) {
+      // the link between the device and the host does not support native
+      // atomic operations
+      Value = UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_ACCESS |
+              UR_DEVICE_USM_ACCESS_CAPABILITY_FLAG_CONCURRENT_ACCESS;
+    }
+    return ReturnValue(Value);
+  }
+
+  case UR_DEVICE_INFO_BACKEND_RUNTIME_VERSION: {
+    int Major = 0, Minor = 0;
+    sycl::detail::ur::assertion(
+        hipDeviceComputeCapability(&Major, &Minor, hDevice->get()) ==
+        hipSuccess);
+    std::string Result = std::to_string(Major) + "." + std::to_string(Minor);
+    return ReturnValue(Result.c_str());
+  }
+
+  case UR_DEVICE_INFO_ATOMIC_64: {
+    // TODO: Reconsider it when AMD supports SYCL_USE_NATIVE_FP_ATOMICS.
```

Comment:
I think this comment is out of date since native atomics now supported? @hdelan

Reply:
Yep comment can be removed. Logic is OK though

Reply:
Fixed